src/plugins/data/public/plugin.ts

```diff
@@ -183,7 +183,7 @@ export class DataPublicPlugin
 
     const useNewSavedQueriesUI =
       core.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED) &&
-      this.config.savedQueriesNewUI.enabled;
+      this.config.savedQueriesNewUI?.enabled;
```

Comment:
why this change?

Reply:
these files aren't covered by any UTs, so when writing tests for data/public/plugin since all these services are initialized, i was getting the error `Cannot read property of undefined` for all these lies [ref: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8798#discussion_r1828538194]

Reply:
Ideally, we're just mocking things to not have to change the source code. It's mainly problematic if we fail in an unexpected way (there are times where failing fast and throwing is preferable to going down a code path that we didn't intend, and patch coverage % points to that being a concern here), but its usages in this CR don't seem that problematic, so I'm fine approving. Please do review however, and if the optional chaining presents a risk, let's fix in tests vs source.